### PR TITLE
Fix/stats type

### PR DESCRIPTION
**Pull request by eldesh**
fix stats api.

# New features

- extend `Stats` to accept values returned from suspended containers
- add `oneshot` option to the stats api

# Improvements

- add a unit test for the stats api
- add fields to the `Stats` type
- update `fixtures/` files
- update type of `Stats::networks` to match the new api

### CODE REVIEW

File: src/stats.rs [hunk 1]
```diff
@@ -98,44 +172,105 @@ pub struct MemoryStat {
...
+    use serde::de::{DeserializeOwned, Deserializer};
+    use serde::ser::{Serialize, Serializer};
+
+    pub mod memory_stats {
+        use super::*;
+
+        #[derive(Debug, Serialize, Deserialize)]
+        #[serde(untagged)]
+        enum Plus1<T> {
+            Item(T),
```
Comment (eldesh) [inline src/stats.rs:238]: `Item` occurred in front of `Empty` to take precedence over `Empty`.

File: src/fixtures/stats_stream.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
..._periods":0,"throttled_time":0}},"precpu_stats":{"cpu_usage":{"total_usage":0,"usage_in_kernelmode":0,"usage_in_usermode":0},"throttling_data":{"periods":0,"throttled_periods":0,"throttled_time":0}},"memory_stats":{"usage":8179712,"max_usage":17543168,"stats":{"active_anon":2711552,"active_file":0,"cache":380928,"dirty":4096,"hierarchical_memory_limit":9223372036854771712,"hierarchical_memsw_limit":0,"inactive_anon":360448,"inactive_file":4096,"mapped_file":348160,"pgfault":2960,"pgmajfault":0,"pgpgin":4909,"pgpgout":4149,"rss":2732032,"rss_huge":0,"total_active_anon":2711552,"total_active_file":0,"total_cache":380928,"total_dirty":4096,"total_inactive_anon":360448,"total_inactive_file":4096,"total_mapped_file":348160,"total_pgfault":2960,"total_pgmajfault":0,"total_pgpgin":4909,"total_pgpgout":4149,"total_rss":2732032,"total_rss_huge":0,"total_unevictable":0,"total_writeback":0,"unevictable":0,"writeback":0},"limit":8340705280},"name":"/dockworker_test_read_stats_container","id":"621d5971f1e21dddbcff975a36a13c44e9f387897708e6831ff95b5cd06a2aae","networks":{"eth0":{"rx_bytes":266,"rx_packets":3,"rx_errors":0,"rx_dropped":0,"tx_bytes":0,"tx_packets":0,"tx_errors":0,"tx_dropped":0}}}
```
Comment (eldesh) [inline src/fixtures/stats_stream.json:1]: raw data for wrapping into the HTTP response body

**Review by nyuichi — APPROVED** [review]
LGTM 💯

**Comment by eldesh** [comment]
:pray:

**Pull request merged**